tests/src/Tests/ErrorsTests.cpp

```diff
@@ -667,6 +667,41 @@ external
 
 		/************************************************************************/
 
+		SECTION("Functions")
+		{
+			CHECK_THROWS_WITH(Compile(R"(
+[nzsl_version("1.0")]
+module;
+
+fn test() -> i32
+{
+	return 42.666;
+}
+)"), "(7,2 -> 15): CUnmatchingTypes error: left expression type (f32) doesn't match right expression type (i32)");
```

Comment:
I think it would be better to use a dedicated error, because TypeMustMatch is not very clear. Also in the future we could add the source location of the return type to be more clear

Reply:
Fixed, I would say maybe source location could be a thing, but I would be worried about the size of the error message in this case, especially since the type is already displayed anyway, and we don't allow things like a function inside another function